sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -73,32 +75,48 @@ protected ServiceBusMessage fromByte(byte[] payload) {
     @Override
     protected void setCustomHeaders(MessageHeaders headers, ServiceBusMessage message) {
 
+        Set<String> copyHeaders = new HashSet<String>();
+        headers.forEach((key, value) -> {
+            copyHeaders.add(key);
+        });
         // Spring MessageHeaders
-        getStringHeader(headers, MessageHeaders.ID).ifPresent(message::setMessageId);
-        getStringHeader(headers, MessageHeaders.CONTENT_TYPE).ifPresent(message::setContentType);
-        getStringHeader(headers, MessageHeaders.REPLY_CHANNEL).ifPresent(message::setReplyTo);
+        getStringHeader(headers, copyHeaders, MessageHeaders.ID).ifPresent(message::setMessageId);
+        getStringHeader(headers, copyHeaders, MessageHeaders.CONTENT_TYPE).ifPresent(message::setContentType);
+        getStringHeader(headers, copyHeaders, MessageHeaders.REPLY_CHANNEL).ifPresent(message::setReplyTo);
 
         // AzureHeaders.
-        getStringHeader(headers, AzureHeaders.RAW_ID).ifPresent(message::setMessageId);
-        Optional.ofNullable(headers.get(AzureHeaders.SCHEDULED_ENQUEUE_MESSAGE, Integer.class))
+        getStringHeader(headers, copyHeaders, AzureHeaders.RAW_ID).ifPresent(message::setMessageId);
+        Optional.of(AzureHeaders.SCHEDULED_ENQUEUE_MESSAGE)
```

Comment:
really wired to see this kind of usage of Optional, please stop it

Reply:
Roll back the original code with the right usage of Optional, and add new operations in the `ifPresent` step.